services/harmony/app/backends/workflow-orchestration/work-item-updates.ts

```diff
@@ -243,6 +243,10 @@ async function handleFailedWorkItems(
             jobMessage = `Maximum allowed errors ${env.maxErrorsForJob} exceeded. See the errors fields for more details`;
             logger.warn(jobMessage);
             continueProcessing = false;
+          } else if (100.0 * errorCount / job.numInputGranules > env.maxPercentErrorsForJob) {
```

Comment:
The intent of the feature is to fail after some smallish number of failures if almost everything is failing rather than getting to 2000 errors.

Something along the lines of:
```
MIN_COMPLETED_GRANULES_TO_CHECK_FAILURE_PERCENTAGE=10
MAX_PERCENT_ERRORS_FOR_JOB=75
```

Then you'd do something like:
```
if (errorCount + successCount > MIN_COMPLETED_GRANULES_TO_CHECK_FAILURE_PERCENTAGE && errorCount / errorCount + successCount > MAX_PERCENT_ERRORS_FOR_JOB) {
  fail job
}
```

I'm not positive how we'd capture the successCount with an inexpensive query.